### PR TITLE
Add support for Composer maintenance window configuration

### PR DESCRIPTION
**Pull request by gor-obr**
Add maintenance_window to google_composer_environment (beta)

fixes hashicorp/terraform-provider-google#9423

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
composer: added field `maintenance_window` to resource `google_composer_environment` (beta)
```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb [hunk 1]
```diff
@@ -52,11 +52,10 @@ var (
 		"config.0.private_environment_config",
 <% unless version == "ga" -%>
 		"config.0.web_server_network_access_control",
-<% end -%>
-<% unless version == "ga" -%>
 		"config.0.database_config",
 		"config.0.web_server_config",
-    "config.0.encryption_config",
+	    "config.0.encryption_config",
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_composer_environment.go.erb:57]: nit: Can you change this to a tab?

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb [hunk 2]
```diff
@@ -1055,6 +1107,13 @@ func expandComposerEnvironmentConfig(v interface{}, d *schema.ResourceData, conf
 	}
 	transformed.EncryptionConfig = transformedEncryptionConfig
 
+	log.Printf("tst tst tst")
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_composer_environment.go.erb:1110]:
```suggestion
```

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb [hunk 3]
```diff
@@ -461,6 +460,36 @@ func resourceComposerEnvironment() *schema.Resource {
 								},
 							},
 						},
+						"maintenance_window": {
+							Type:         schema.TypeList,
+							Optional:     true,
+							Computed:     false,
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_composer_environment.go.erb:466]: The failing tests indicate that this field is set by the API even when not set by the user. I think we want this to be `Computed: true`
```suggestion
							Computed:      true,
```

**Review by rileykarson — COMMENTED** [review]
Hmm- looks like some tests are failing, with an error like below. See inline.

```
------- Stdout: -------
=== RUN   TestAccComposerEnvironment_basic
=== PAUSE TestAccComposerEnvironment_basic
=== CONT  TestAccComposerEnvironment_basic
provider_test.go:276: Step 1/4 error: After applying this test step, the plan was not empty.
stdout:
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
~ update in-place
Terraform will perform the following actions:
# google_composer_environment.test will be updated in-place
~ resource "google_composer_environment" "test" {
id      = "projects/ci-test-project-188019/locations/us-central1/environments/tf-test-composer-env-4732599840812260542"
name    = "tf-test-composer-env-4732599840812260542"
# (2 unchanged attributes hidden)
~ config {
# (4 unchanged attributes hidden)
- maintenance_window {
- end_time   = "1970-01-01T00:00:00Z" -> null
- start_time = "1970-01-01T00:00:00Z" -> null
}
~ node_config {
+ tags                 = []
# (9 unchanged attributes hidden)
}
# (6 unchanged blocks hidden)
}
}
Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccComposerEnvironment_basic (1312.50s)
FAIL
```

**Comment by rileykarson** [comment]
(oh whoops, not sure why I started taking a look at this given that the bot assigned @melinath. Sorry about that @melinath, I can continue the review!)

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb (hunk 1 above)
Comment (gor-obr) [reply mmv1/third_party/terraform/resources/resource_composer_environment.go.erb]: done.

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb (hunk 2 above)
Comment (gor-obr) [reply mmv1/third_party/terraform/resources/resource_composer_environment.go.erb]: done.

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb (hunk 3 above)
Comment (gor-obr) [reply mmv1/third_party/terraform/resources/resource_composer_environment.go.erb]: done.

**Review by rileykarson — APPROVED** [review]

**Pull request merged**